File: docs/sdks/js/example-list-buckets.js
'use strict'

var AWS = require('aws-sdk');

var config = {
  accessKeyId: "YOUR_MINIO_ACCESS_ID",
  secretAccessKey: "YOUR_MINIO_SECRET_KEY",
  endpoint: "localhost:9000",
  region: "",
  sslEnabled: false
};

AWS.config.update(config);

var s3 = new AWS.S3();
s3.listBuckets(function(err, data) {
  if (err) {
    console.log(err); // an error occurred
  } else {
    console.log(data); // successful response
  }
});

File: docs/sdks/ruby/example-list-buckets.rb
#!/usr/bin/env ruby
require 'aws-sdk'

s3 = Aws::S3::Client.new(endpoint: "http://localhost:9000/",
                         require_https_for_sse_cpk: false,
                         region: "minio")
resp = s3.list_buckets
resp.buckets.each do |bucket|
  puts "#{bucket.name} => #{bucket.creation_date}"
end
